packages/medusa/src/services/draft-order.ts

```diff
@@ -309,34 +310,39 @@ class DraftOrderService extends TransactionBaseService {
         const itemsToCreate: Partial<LineItem>[] = []
 
         // prepare that for next steps
-        ;(items ?? []).forEach((item) => {
-          if (item.variant_id) {
-            itemsToGenerate.push({
-              variantId: item.variant_id,
+        await Promise.all(
+          (items ?? []).map(async (item) => {
+            if (item.variant_id) {
+              const variant = (await this.productVariantService_.retrieve(
```

Comment:
todo: to prevent reducing the performance, I would suggest to list the variant before that are in the ids of the items.variant_id and create a map of it. Then you can just do the usual map to prepare and generate